### PR TITLE
fix(regex): fix data file for 942370

### PR DESCRIPTION
**Pull request by emphazer**
disassemble regex in rule 942370

### CODE REVIEW

**Review by fzipi — DISMISSED** [review]
LGTM

File: util/regexp-assemble/data/942370.data [hunk 1]
```diff
@@ -3,22 +3,44 @@
...
-[\"'`]\s*?\*.+like\W*?[\"'`]\d
-[\"'`]\s*?\*.+between\W*?[\"'`]\d
-[\"'`]\s*?\*.+and\W*?[\"'`]\d
-[\"'`]\s*?\*.+id\W*?[\"'`]\d
+##!> assemble
+[\"'`]\s*?
+##!=>
+ ##!> assemble
+  ##!> assemble
+  ##!=>
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:11]:
```suggestion
```

File: util/regexp-assemble/data/942370.data [hunk 2]
```diff
@@ -3,22 +3,44 @@
...
 
-[\"'`]\s*?\*.+or\W*?[\"'`]\d
-[\"'`]\s*?\*.+xor\W*?[\"'`]\d
-[\"'`]\s*?\*.+div\W*?[\"'`]\d
-[\"'`]\s*?\*.+like\W*?[\"'`]\d
-[\"'`]\s*?\*.+between\W*?[\"'`]\d
-[\"'`]\s*?\*.+and\W*?[\"'`]\d
-[\"'`]\s*?\*.+id\W*?[\"'`]\d
+##!> assemble
+[\"'`]\s*?
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:7]: Use a template for ```[\"'`]```, as its repeated multiple times.

File: util/regexp-assemble/data/942370.data [hunk 3]
```diff
@@ -3,22 +3,44 @@
...
+##!=>
+ ##!> assemble
+  ##!> assemble
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:10]: You are using two nested assemble blocks here, but you only need one, the outer one doesn't do anything.

File: util/regexp-assemble/data/942370.data [hunk 4]
```diff
@@ -3,22 +3,44 @@
...
+[^\w\s?]+\s*?[^\w\s]+\s*?[\"'`]
+[^\w\s]+\s*?[\W\d].*?#
+[^\w\s]+\s*?[\W\d].*?--
+   ##!> assemble
+or
+xor
+div
+like
+between
+and
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:36]: You can store the list of words that is repeated (or, xor, div, etc.) and reuse it, so you don't have to write it down twice.

**Review by theseion — CHANGES_REQUESTED** [review]

File: util/regexp-assemble/data/942370.data [hunk 5]
```diff
@@ -3,22 +3,45 @@
...
+##!=< operators
+##!<
+
+##! Main assembly
+##!> assemble
+{{quotes}}\s*?
+##!=>
+ ##!> assemble
+\*.+
+  ##!=>
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:25]:
```suggestion
 ##!=>
```

File: util/regexp-assemble/data/942370.data [hunk 6]
```diff
@@ -3,22 +3,45 @@
...
+   ##!> assemble
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:26]: Indent this block by four spaces, not 3.

File: util/regexp-assemble/data/942370.data [hunk 7]
```diff
@@ -3,22 +3,45 @@
...
+   ##!<
+id
+  ##!=>
+\W*?{{quotes}}\d
+  ##!=>
+  ##!<
+[^\w\s?]+\s*?[^\w\s]+\s*?{{quotes}}
+[^\w\s]+\s*?[\W\d].*?#
+[^\w\s]+\s*?[\W\d].*?--
+   ##!> assemble
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:37]:
```suggestion
```

File: util/regexp-assemble/data/942370.data [hunk 8]
```diff
@@ -3,22 +3,45 @@
...
+   ##!=> operators
+  ##!=>
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:39]:
```suggestion
```

File: util/regexp-assemble/data/942370.data [hunk 9]
```diff
@@ -3,22 +3,45 @@
...
+\s[^\d]+[\w-]+.*?\d
+  ##!=>
+ ##!<
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:42]:
```suggestion
```

File: util/regexp-assemble/data/942370.data [hunk 10]
```diff
@@ -3,22 +3,45 @@
...
+  ##!=>
+  ##!<
+[^\w\s?]+\s*?[^\w\s]+\s*?{{quotes}}
+[^\w\s]+\s*?[\W\d].*?#
+[^\w\s]+\s*?[\W\d].*?--
+   ##!> assemble
+   ##!=> operators
+  ##!=>
+\s[^\d]+[\w-]+.*?\d
+  ##!=>
```
Comment (theseion) [inline util/regexp-assemble/data/942370.data:41]:
```suggestion
```

**Review by theseion — CHANGES_REQUESTED** [review]
I think we need to clean up some more.

**Comment by fzipi** [comment]
@theseion Force pushed a new rebase version. Please double-check.

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
@fzipi Added tests and fixed assembly. Please review.

**Review by fzipi — APPROVED** [review]
This looks awesome now. I agree that this should be part of the big documentation project for v5 maybe: knowing what the rules are testing and why.

**Pull request merged**